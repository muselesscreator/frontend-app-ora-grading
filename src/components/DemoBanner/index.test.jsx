import React from 'react';
import { shallow } from 'enzyme';

import { DemoBanner } from '.';

describe('DemoBanner component', () => {
  let el;
  describe('snapshots', () => {
    const OLD_ENV = process.env;
    beforeAll(() => {
      jest.resetModules();
      process.env = { ...OLD_ENV };
    });
    afterAll(() => {
      process.env = OLD_ENV;
    });
    test('does not render if disabled flag is missing', () => {
      process.env.REACT_APP_NOT_ENABLED = false;
      el = shallow(<DemoBanner />);
      expect(el).toMatchSnapshot();
      expect(el.isEmptyRender()).toEqual(true);
    });
    test('snapshot: disabled flag is present', () => {
      process.env.REACT_APP_NOT_ENABLED = true;
      el = shallow(<DemoBanner />);
      expect(el).toMatchSnapshot();
      expect(el.isEmptyRender()).toEqual(false);
    });
  });
});
