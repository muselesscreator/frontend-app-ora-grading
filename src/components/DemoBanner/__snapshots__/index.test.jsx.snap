// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DemoBanner component snapshots does not render if disabled flag is missing 1`] = `""`;

exports[`DemoBanner component snapshots snapshot: disabled flag is present 1`] = `
<Component>
  <FormattedMessage
    defaultMessage="You are using a demo mode of the new ORA Staff Grading Experience!"
    description="Inform user for using demo mod for ORA Staff Grading Experiences"
    id="ora-grading.DemoBanner.demoBannerMessage"
  />
</Component>
`;
